### PR TITLE
ci: add go1.17

### PR DESCRIPTION
**Pull request by kolyshkin**
Keeping this as draft until go1.17 is released. Will rebase periodically to test master with go 1.17.

* Jul 13: updated to rc1.
* Aug 3: updated to rc2.
* Aug 16: updated to GA.

### CODE REVIEW

**Comment by kolyshkin** [comment]
CI failure is unrelated (https://github.com/opencontainers/runc/issues/3050)

**Comment by kolyshkin** [comment]
Rebased.

**Comment by kolyshkin** [comment]
Updated to 1.17-rc1

**Comment by kolyshkin** [comment]
Removed go 1.15

**Comment by AkihiroSuda** [comment]
We have 1.17 GA now https://golang.org/dl/

**Comment by kolyshkin** [comment]
Updated to GA; no longer a draft.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go-version: [1.15.x, 1.16.x]
+        go-version: [1.15.x, 1.16.x, 1.17.x]
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:20]: Dockerfile needs to be update https://github.com/opencontainers/runc/blob/55e93b89f185631a87d87b9f4e7e08437357eb2c/Dockerfile#L1

Comment (AkihiroSuda) [reply .github/workflows/test.yml:20]: Can we drop support for Go 1.15?

Comment (kolyshkin) [reply .github/workflows/test.yml:20]: > Dockerfile needs to be update

done

> Can we drop support for Go 1.15?

I'd say let's have it for some time as it just became unsupported a day ago.

This is also a separate issue; I'd like to add 1.17 now and drop 1.15 later and separately.

**Review by AkihiroSuda — APPROVED** [review]

**Review by mrunalp — APPROVED** [review]

**Pull request merged**